src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts

```diff
@@ -1,27 +1,245 @@
+    feedbackSessionData: testFeedbackSession,
+    feedbackSessionLogEntries: [
+      {
+        studentData: testStudent,
+        feedbackSessionLogType: LogType.FEEDBACK_SESSION_SUBMISSION,
+        timestamp: 0,
+      },
+    ],
+  };
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       imports: [InstructorAuditLogsPageModule, HttpClientTestingModule],
     })
-    .compileComponents();
+        .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(InstructorAuditLogsPageComponent);
+    courseService = TestBed.inject(CourseService);
+    studentService = TestBed.inject(StudentService);
+    logService = TestBed.inject(LogService);
+    timezoneService = TestBed.inject(TimezoneService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should snap with default fields', () => {
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when page is still loading', () => {
+    component.isLoading = true;
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap when searching for details in search form', () => {
+    component.courses = [testCourse1, testCourse2];
+    component.formModel = {
+      logsDateFrom: { year: 1997, month: 9, day: 11 },
+      logsTimeFrom: { hour: 23, minute: 59 },
+      logsDateTo: { year: 1998, month: 9, day: 11 },
+      logsTimeTo: { hour: 15, minute: 0 },
+      courseId: 'CS9999',
+      studentName: 'Doe John',
+    };
+    component.courseToStudents = {
+      CS9999: [testStudent],
+      MA1234: [],
+    };
+    component.isLoading = false;
+    component.isSearching = true;
+    fixture.detectChanges();
+
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with results of a search', () => {
+    component.searchResults = [
+      {
+        feedbackSessionName: 'Feedback session 1',
+        logColumnsData: resultColumns,
+        logRowsData: [[
+          { value: '15 January 2021' },
+          { value: 'Doe John' },
+          { value: 'Viewed the submission page' },
+          { value: 'doejohn@email.com' },
+          { value: 'section 1' },
+          { value: 'team 1' },
+        ]],
+        isTabExpanded: true,
+      },
+      {
+        feedbackSessionName: 'Feedback session 2',
+        logColumnsData: resultColumns,
+        logRowsData: [],
+        isTabExpanded: true,
+      },
+    ];
+    component.isLoading = false;
+    component.isSearching = false;
+    fixture.detectChanges();
+
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should load all courses and students that instructor has on init', () => {
+    const emptyStudent: Student = {
+      courseId: '', email: '', name: '', sectionName: '', teamName: '',
+    };
+    const courseSpy: Spy = spyOn(courseService, 'getAllCoursesAsInstructor').and
```

Comment:
same for all these *Spy variables, should we declare them globally since I think they might be used in many future test cases ?

Reply:
I think the Spy variables should be declared locally for now since there is no common place it is being used. They are usually unique to the test case in their return values so not many test cases will be re-using the spy.

Reply:
that makes sense, lets leave them as they are for now